### ISSUE TITLE
Nexus 5 official ROM Android 5.1.1 Crash

### ISSUE THREAD

**Issue by xax007** [shadowsocks/shadowsocks-android#2003]
After adb install, open shadowsocks, alert `Unfortunately, shadowsocks has stoped.`

![](https://ws2.sinaimg.cn/large/006tNbRwgy1fx8mjmefxoj30u01hc7b2.jpg)

**Smartphone :**
 - Android version: 5.1.1 official ROM
 - Device: Google Nexus 5
 - Version: shadowsocks-armeabi-v7a-4.6.1.apk

**Additional context**
`adb logcat`

```
I/ActivityManager(  745): START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.github.shadowsocks/.MainActivity (has extras)} from uid 10022 on display 0
V/WindowManager(  745): addAppToken: AppWindowToken{18606ce6 token=Token{24afbf41 ActivityRecord{2e57c628 u0 com.github.shadowsocks/.MainActivity t33}}} to stack=1 task=33 at 0
D/audio_hw_primary(  191): out_set_parameters: enter: usecase(1: low-latency-playback) kvpairs: routing=2
V/WindowManager(  745): Adding window Window{97c1579 u0 Starting com.github.shadowsocks} at 3 of 10 (after Window{d6ed460 u0 com.android.chrome/com.google.android.apps.chrome.document.DocumentActivity})
D/audio_hw_primary(  191): select_devices: out_snd_device(2: speaker) in_snd_device(0: none)
D/msm8974_platform(  191): platform_send_audio_calibration: sending audio calibration for snd_device(2) acdb_id(15)
D/audio_hw_primary(  191): enable_snd_device: snd_device(2: speaker)
D/audio_hw_primary(  191): enable_audio_route: apply and update mixer path: low-latency-playback
W/art     (24671): Before Android 4.1, method android.graphics.PorterDuffColorFilter android.support.graphics.drawable.VectorDrawableCompat.updateTintFilter(android.graphics.PorterDuffColorFilter, android.content.res.ColorStateList, android.graphics.PorterDuff$Mode) would have incorrectly overridden the package-private method in android.graphics.drawable.Drawable
D/AndroidRuntime(24671): Shutting down VM
E/AndroidRuntime(24671): FATAL EXCEPTION: main
E/AndroidRuntime(24671): Process: com.github.shadowsocks, PID: 24671
E/AndroidRuntime(24671): java.lang.RuntimeException: Unable to start activity ComponentInfo{com.github.shadowsocks/com.github.shadowsocks.MainActivity}: java.lang.RuntimeException: Could not inflate Behavior subclass com.github.shadowsocks.widget.ShrinkUpwardBehavior
E/AndroidRuntime(24671): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2325)
E/AndroidRuntime(24671): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2387)
E/AndroidRuntime(24671): 	at android.app.ActivityThread.access$800(ActivityThread.java:151)
E/AndroidRuntime(24671): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1303)
E/AndroidRuntime(24671): 	at android.os.Handler.dispatchMessage(Handler.java:102)
E/AndroidRuntime(24671): 	at android.os.Looper.loop(Looper.java:135)
E/AndroidRuntime(24671): 	at android.app.ActivityThread.main(ActivityThread.java:5254)
E/AndroidRuntime(24671): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(24671): 	at java.lang.reflect.Method.invoke(Method.java:372)
E/AndroidRuntime(24671): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
E/AndroidRuntime(24671): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
E/AndroidRuntime(24671): Caused by: java.lang.RuntimeException: Could not inflate Behavior subclass com.github.shadowsocks.widget.ShrinkUpwardBehavior
E/AndroidRuntime(24671): 	at android.support.design.widget.CoordinatorLayout.parseBehavior(CoordinatorLayout.java:622)
E/AndroidRuntime(24671): 	at android.support.design.widget.CoordinatorLayout$LayoutParams.<init>(CoordinatorLayout.java:2805)
E/AndroidRuntime(24671): 	at android.support.design.widget.CoordinatorLayout.generateLayoutParams(CoordinatorLayout.java:1704)
E/AndroidRuntime(24671): 	at android.support.design.widget.CoordinatorLayout.generateLayoutParams(CoordinatorLayout.java:111)
E/AndroidRuntime(24671): 	at android.view.LayoutInflater.rInflate(LayoutInflater.java:808)
E/AndroidRuntime(24671): 	at android.view.LayoutInflater.inflate(LayoutInflater.java:504)
E/AndroidRuntime(24671): 	at android.view.LayoutInflater.inflate(LayoutInflater.java:414)
E/AndroidRuntime(24671): 	at android.view.LayoutInflater.inflate(LayoutInflater.java:365)
E/AndroidRuntime(24671): 	at android.support.v7.app.AppCompatDelegateImpl.setContentView(AppCompatDelegateImpl.java:467)
E/AndroidRuntime(24671): 	at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:140)
E/AndroidRuntime(24671): 	at com.github.shadowsocks.MainActivity.onCreate(MainActivity.kt:218)
E/AndroidRuntime(24671): 	at android.app.Activity.performCreate(Activity.java:5990)
E/AndroidRuntime(24671): 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1106)
E/AndroidRuntime(24671): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2278)
E/AndroidRuntime(24671): 	... 10 more
E/AndroidRuntime(24671): Caused by: java.lang.reflect.InvocationTargetException
E/AndroidRuntime(24671): 	at java.lang.reflect.Constructor.newInstance(Native Method)
E/AndroidRuntime(24671): 	at java.lang.reflect.Constructor.newInstance(Constructor.java:288)
E/AndroidRuntime(24671): 	at android.support.design.widget.CoordinatorLayout.parseBehavior(CoordinatorLayout.java:620)
E/AndroidRuntime(24671): 	... 23 more
E/AndroidRuntime(24671): Caused by: kotlin.KotlinNullPointerException
E/AndroidRuntime(24671): 	at com.github.shadowsocks.widget.ShrinkUpwardBehavior.<init>(ShrinkUpwardBehavior.kt:69)
E/AndroidRuntime(24671): 	... 26 more
W/ActivityManager(  745):   Force finishing activity 1 com.github.shadowsocks/.MainActivity
I/OpenGLRenderer(  745): Initialized EGL, version 1.4
W/ActivityManager(  745): Activity pause timeout for ActivityRecord{2e57c628 u0 com.github.shadowsocks/.MainActivity t33 f}
```

**Comment by Mygod** [comment]
This will be fixed in the next release.

**Comment by Mygod** [comment]
This should be fixed in 4.6.2.
